### DISCUSSION TITLE
Support for a `z.example()` method

### DISCUSSION THREAD

**Discussion by erunion** [colinhacks/zod#2266] [Ideas]
We're open to PR'ing this in but before we do I'm curious if there's interest in a `.example()` method that like `.describe()` adds an `example` property into the `ZodTypeDef` interface so that tooling like `zod-to-json-schema` could generate JSON Schema representations with examples.

For example:

```js
const documentedURL = z
  .url()
  .example('https://example.com')
documentedURL.example; // https://example.com
```

**Comment by ryceg** [comment]
This would be a great feature.

**Comment by erunion** [comment]
I have added support for this in https://github.com/colinhacks/zod/pull/3518.

**Comment by erunion** [comment]
This looks to now be supported in the new `.meta()` work in v4!

https://v4.zod.dev/v4#metadata
